src/Agent.Plugins/PipelineCache/FingerprintCreator.cs

```diff
@@ -0,0 +1,244 @@
+﻿using Agent.Sdk;
+using BuildXL.Cache.ContentStore.Interfaces.Utils;
+using Microsoft.VisualStudio.Services.BlobStore.Common;
+using Microsoft.VisualStudio.Services.PipelineCache.WebApi;
+using Minimatch;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+
+[assembly: InternalsVisibleTo("Test")]
+
+namespace Agent.Plugins.PipelineCache
+{
+    public static class FingerprintCreator
+    {
+        private static readonly bool isWindows = Helpers.IsWindowsPlatform(Environment.OSVersion);
+
+        // https://github.com/Microsoft/azure-pipelines-task-lib/blob/master/node/docs/findingfiles.md#matchoptions
+        private static readonly Options minimatchOptions = new Options
+        {
+            Dot = true,
+            NoBrace = true,
+            NoCase = isWindows,
```

Comment:
@chrispat @bryanmacfarlane  @willsmythe We should be case-insensitive on OSX, too - right?